src/theory/arith/arith_utilities.h

```diff
@@ -338,6 +338,13 @@ Node negateProofLiteral(TNode n);
  */
 Node multConstants(const Node& c1, const Node& c2);
 
+/**
+ * Make the equality (= a b) or (= (- a b) zero) if a and b have different
+ * types, where zero has the same type as (- a b).
+ * Use this utility to ensure an equality is properly typed.
+ */
+Node mkEquality(Node a, Node b);
```

Comment:
I assume this could change solver behavior, given that the equalities look different now for mixed integer?

Reply:
This shouldn't change anything (yet), since the places that use this utility have their equalities rewritten.  I believe (= a b) and (= (- a b) 0) should in almost all cases rewrite to the same thing currently.

The tricky issue will be if it is important that the sides of the equality are maintained; this is the case for theory combination and the equality engine.  Neither of these I have addressed yet.